### PR TITLE
Implement Clone for Queue

### PR DESCRIPTION
**Pull request by XOSplicer**
ref #56

### CODE REVIEW

**Comment by XOSplicer** [comment]
bors retry

**Comment by XOSplicer** [comment]
Sorry for the merge conflicts

File: src/spsc/mod.rs [hunk 1]
```diff
@@ -394,6 +394,30 @@ macro_rules! impl_ {
...
         }
+
+        impl<T, N, C> Clone for Queue<T, N, $uxx, C>
+        where
+            T: Clone,
+            N: ArrayLength<T>,
+            C: sealed::XCore,
+        {
+            fn clone(&self) -> Self {
+                let mut new = Queue {
```
Comment (japaric) [inline src/spsc/mod.rs]: I think you'll need to specify the full type here (`let mut new: Queue<..> = ..`), otherwise method lookup will fail below.

**Comment by japaric** [comment]
bors delegate+

You can r+ this after tests are fixed.

File: src/spsc/mod.rs (hunk 1 above)
Comment (XOSplicer) [reply src/spsc/mod.rs]: Yes, this does indeed solve the problem 👍

**Pull request merged**